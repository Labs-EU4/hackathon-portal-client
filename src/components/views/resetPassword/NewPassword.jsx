import React from "react";
import { Formik, Form, ErrorMessage } from "formik";
import * as Yup from "yup";
import { useDispatch } from "react-redux";
import Input from "../../../assets/atoms/Input";
import WideBody from "../../../assets/atoms/WideBody";
import BodyContainer from "../../../assets/atoms/BodyContainer";
import HeroImage from "../../../assets/atoms/HeroImage";
import image from "../../../assets/imagesSignup.png";
import Container from "../../../assets/atoms/Container";
import { H1 } from "../../../assets/atoms/Heading";
import { Paragraph } from "../../../assets/atoms/Paragraph";
import { RowBody } from "../../../assets/atoms/RowBody";
import { Header, Footer } from "../../organisms/index";
import Button from "../../atoms/Button";
import { ErrorSpan } from "../../atoms/Span";
import { resetPassword } from "../../../store/user/actions";
import { useHistory } from "react-router-dom";

const ResetPassword = () => {
  const dispatch = useDispatch();
  const history = useHistory();

  const handleSubmit = values => {
    const { newPassword, newPasswordConfirm } = values;
    if (newPassword === newPasswordConfirm) {
      dispatch(resetPassword(newPassword, history));
    }
  };

  const schema = Yup.object().shape({
    newPassword: Yup.string()
      .required("Password is required.")
      .min(8, "Password must be at least 8 characters long."),
    newPasswordConfirm: Yup.string()
      .oneOf([Yup.ref("newPassword"), null], "Passwords must match")
      .required("Password confirm is required.")
  });

  return (
    <div>
      <Header />
      <WideBody>
        <BodyContainer justify="center">
          <HeroImage src={image} alt="Reset passowrd" />
          <Container>
            <H1>Change the password</H1>
            <Paragraph>
              Please enter new password and confirm it in the second input
              field.
            </Paragraph>
            <Formik
              initialValues={{ newPassword: "", newPasswordConfirm: "" }}
              onSubmit={handleSubmit}
              validationSchema={schema}
            >
              {({ errors, touched }) => (
                <Form>
                  <Input
                    display="wide"
                    id="newPassword"
                    type="password"
                    name="newPassword"
                    placeholder="Enter new password"
                  />
                  {errors.name && touched.name ? (
                    <div>{errors.name}</div>
                  ) : null}
                  <ErrorSpan>
                    <ErrorMessage name="newPassword" />
                  </ErrorSpan>
                  <Input
                    display="wide"
                    id="newPasswordConfirm"
                    type="password"
                    name="newPasswordConfirm"
                    placeholder="Confirm new password"
                  />
                  {errors.name && touched.name ? (
                    <div>{errors.name}</div>
                  ) : null}
                  <ErrorSpan>
                    <ErrorMessage name="newPasswordConfirm" />
                  </ErrorSpan>
                  <RowBody>
                    <Button type="submit" size="wide" color="blue">Change Password</Button>
                  </RowBody>
                </Form>
              )}
            </Formik>
          </Container>
        </BodyContainer>
      </WideBody>
      <Footer />
    </div>
  );
};

export default ResetPassword;
