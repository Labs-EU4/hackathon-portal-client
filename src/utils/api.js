import Axios from "axios";
import { toast } from "react-toastify";
import { resetUser } from "../store/user/actions";

const baseURL = process.env.REACT_APP_API_URL;

// if process.env.CONTEXT = "production" {
//   Point to heroku API
// }
// if process.env.BRANCH = "develop" {
  // point to Heroku staging API
// }
// if process.env.CONTEXT = "deploy-preview" {
//   point to deploy-preview HEROKU API (the separate one)
// }



export const axiosWithAuth = token =>
  Axios.create({
    baseURL,
    headers: {
      Authorization: token
    }
  });

export const axios = Axios.create({ baseURL });

export const selectToken = state => state.currentUser.token;

export const showError = message => {
  toast.error(message);
};

export const showSuccess = message => {
  toast.success(message);
};

export function* handleError({ response }, put, history) {
  const { message, statusCode } = response.data;
  if (statusCode === 404) {
    history.push("/not-found");
  }
  if (message === "jwt expired") {
    yield put(resetUser());
  } else showError(`⚠️ ${message}`);
}
