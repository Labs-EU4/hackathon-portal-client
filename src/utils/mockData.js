export const initialState = {
  events: {
    data: [
      {
        id: 1,
        event_title: "EuroHack",
        event_description: "the best event hack hack hack macbook",
        creator_id: 1,
        start_date: "10/09/11",
        end_date: "08/07/13",
        location: "Europe",
        guidelines: "cool hekhkdsjfhsdkjf skdjfhskdjfhkjsdfh skjdfhksjdfh",
        participation_type: "Team",
        category_id: 1,
        requirements: [
          {
            video_url:
              "https://res.cloudinary.com/ibkrice-com/video/upload/v1581259780/Harvesting_ifato8.mp4"
          }
        ],
        ProfileImage: [
          {
            image_url:
              "https://res.cloudinary.com/ibkrice-com/video/upload/v1581259780/Harvesting_ifato8.mp4"
          }
        ],
        rubrics: [
          "Presentation",
          "Product Fit",
          "Functionality",
          "Innovation",
          "Product Design",
          "Extensibility"
        ]
      },
      {
        id: 2,
        event_title: "World",
        event_description: "number one hacker event in the world",
        creator_id: 1,
        start_date: "07/09/11",
        end_date: "08/07/13",
        location: "Asia",
        guidelines: "come well prepared,no cheating,no fighting",
        participation_type: "Team",
        category_id: 2
      }
    ],
    categories: [
      { id: 2, category_name: "Summer Hackton" },
      { id: 3, category_name: "World Hackton" },
      { id: 3, category_name: "Asia Hackton" },
      { id: 3, category_name: "Gamer's Hackton" },
      { id: 3, category_name: "Advanced Hackton" }
    ]
  },

  currentUser: {
    token: 347534574536,
    userId: 1,
    fullname: "Jake Tom",
    email: "8omemail@google.com",
    username: "Jake22",
    bio: "I like coding ",
    image: ""
  },
  submissions: [
    {
      id: 1,
      project_title: "Mock Project",
      participant_or_team_name: "Mock Team",
      event_id: 7,
      video_url: "",
      git_url: "https://www.google.com/",
      project_writeups: "No mock write up",
      submitted_by: 14,
      created_at: "2020-03-02",
      updated_at: "2020-03-02",
      average_rating: 0,
      acted_judges: 0,
      number_of_judges: 0
    }
  ],

  teams: [
    { id: 1, team_name: "The A team", team_lead: 2 },
    { id: 2, team_name: "Hacking central", team_lead: 3 },
    { id: 3, team_name: "blackhats", team_lead: 4 },
    { id: 4, team_name: "Spies", team_lead: 1 }
  ],
  participant_team_members: [
    {
      team_member: 3,
      team_id: 1
    },
    {
      team_member: 4,
      team_id: 1
    },
    {
      team_member: 5,
      team_id: 1
    }
  ]
};
