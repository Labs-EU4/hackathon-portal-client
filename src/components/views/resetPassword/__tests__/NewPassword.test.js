import React from "react";
import { Router } from "react-router";
import { createMemoryHistory } from "history";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import configureStore from "redux-mock-store";
import "@testing-library/jest-dom/extend-expect";
import NewPassword from "../NewPassword";
import { initialState } from "../../../../utils/mockData";

const history = createMemoryHistory();

let component;
let mockStore;
let store;

beforeEach(() => {
  mockStore = configureStore();
  store = mockStore(initialState);
  component = render(
    <Router history={history}>
      <Provider store={store}>
        <NewPassword />
      </Provider>
    </Router>
  );
});

describe("Asserts that text nodes on NewPassword.js,render properly", () => {
  it("ResetPassword.js component renders properly", () => {
    expect(component).toMatchSnapshot();
  });
  it("The text node for the <H1> Change the password, renders properly ", () => {
    expect(component.queryByText(/Change the password/i)).toBeInTheDocument();
  });
  //   it("The text node for the  email confirmation, renders properly ", () => {
  //     expect(
  //       component.queryByText(
  //         "Please check your email inbox for directions on how to reset your password."
  //       )
  //     ).toBeInTheDocument();
  //   });
});
