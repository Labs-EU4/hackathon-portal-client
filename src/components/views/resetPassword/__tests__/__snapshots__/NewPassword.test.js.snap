// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asserts that text nodes on NewPassword.js,render properly ResetPassword.js component renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="sc-hzDkRC dKOuJG"
        >
          <div
            class="sc-fBuWsC jWlczg"
          >
            <a
              href="/"
            >
              <img
                alt="Hackton - Organise hackathons"
                class="sc-dVhcbM fIwlsM"
                src="Hackton-logo.png"
              />
            </a>
            <div
              class="sc-eqIVtm gCOrvJ"
            >
              <a
                class="sc-cMljjf lajdkL"
                href="/login"
              >
                Log In
              </a>
              <a
                class="sc-cMljjf dYptiY"
                color="blue"
                href="/register"
              >
                Sign Up
              </a>
            </div>
          </div>
        </div>
        <div
          class="sc-bwzfXH hQwRux"
        >
          <div
            class="sc-bxivhb eJIpFO"
          >
            <img
              alt="Reset passowrd"
              class="sc-EHOje gVUAzz"
              src="Signup.png"
            />
            <div
              class="sc-bZQynM bYrUTg"
            >
              <h1
                class="sc-htoDjs hOpwJQ"
              >
                Change the password
              </h1>
              <p
                class="sc-jzJRlG eWJCf"
              >
                Please enter new password and confirm it in the second input field.
              </p>
              <form
                action="#"
              >
                <input
                  class="sc-bdVaJa kVWpri"
                  display="wide"
                  id="newPassword"
                  name="newPassword"
                  placeholder="Enter new password"
                  type="password"
                  value=""
                />
                <span
                  class="sc-jAaTju sc-iRbamj eBAQsS"
                />
                <input
                  class="sc-bdVaJa kVWpri"
                  display="wide"
                  id="newPasswordConfirm"
                  name="newPasswordConfirm"
                  placeholder="Confirm new password"
                  type="password"
                  value=""
                />
                <span
                  class="sc-jAaTju sc-iRbamj eBAQsS"
                />
                <div
                  class="sc-chPdSV hcWdwp"
                >
                  <button
                    class="sc-brqgnP krIjdA"
                    color="blue"
                    type="submit"
                  >
                    Change Password
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
        <div
          class="sc-kGXeez cXSUJN"
        >
          <div
            class="sc-dxgOiQ nQsul"
          >
            <div>
              <img
                alt="LinkedIn"
                class="sc-jKJlTe cxGEaI"
                src="Icon-linkedin.png"
              />
              <img
                alt="Twitter"
                class="sc-jKJlTe cxGEaI"
                src="Icon-twitter.png"
              />
              <img
                alt="Facebook"
                class="sc-jKJlTe cxGEaI"
                src="Icon-facebook.png"
              />
            </div>
            <div>
              <a
                class="sc-kEYyzF sc-iAyFgw bHoBLw"
                href="https://hackton.co/#team"
              >
                Team
              </a>
              <a
                class="sc-kEYyzF sc-iAyFgw bHoBLw"
                href="#"
              >
                Privacy
              </a>
              <a
                class="sc-kEYyzF sc-iAyFgw bHoBLw"
                href="#"
              >
                Contacts
              </a>
            </div>
            <div>
              <p
                class="sc-jzJRlG sc-cSHVUG iodlNy"
              >
                International Crafters © 2020
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="sc-hzDkRC dKOuJG"
      >
        <div
          class="sc-fBuWsC jWlczg"
        >
          <a
            href="/"
          >
            <img
              alt="Hackton - Organise hackathons"
              class="sc-dVhcbM fIwlsM"
              src="Hackton-logo.png"
            />
          </a>
          <div
            class="sc-eqIVtm gCOrvJ"
          >
            <a
              class="sc-cMljjf lajdkL"
              href="/login"
            >
              Log In
            </a>
            <a
              class="sc-cMljjf dYptiY"
              color="blue"
              href="/register"
            >
              Sign Up
            </a>
          </div>
        </div>
      </div>
      <div
        class="sc-bwzfXH hQwRux"
      >
        <div
          class="sc-bxivhb eJIpFO"
        >
          <img
            alt="Reset passowrd"
            class="sc-EHOje gVUAzz"
            src="Signup.png"
          />
          <div
            class="sc-bZQynM bYrUTg"
          >
            <h1
              class="sc-htoDjs hOpwJQ"
            >
              Change the password
            </h1>
            <p
              class="sc-jzJRlG eWJCf"
            >
              Please enter new password and confirm it in the second input field.
            </p>
            <form
              action="#"
            >
              <input
                class="sc-bdVaJa kVWpri"
                display="wide"
                id="newPassword"
                name="newPassword"
                placeholder="Enter new password"
                type="password"
                value=""
              />
              <span
                class="sc-jAaTju sc-iRbamj eBAQsS"
              />
              <input
                class="sc-bdVaJa kVWpri"
                display="wide"
                id="newPasswordConfirm"
                name="newPasswordConfirm"
                placeholder="Confirm new password"
                type="password"
                value=""
              />
              <span
                class="sc-jAaTju sc-iRbamj eBAQsS"
              />
              <div
                class="sc-chPdSV hcWdwp"
              >
                <button
                  class="sc-brqgnP krIjdA"
                  color="blue"
                  type="submit"
                >
                  Change Password
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
      <div
        class="sc-kGXeez cXSUJN"
      >
        <div
          class="sc-dxgOiQ nQsul"
        >
          <div>
            <img
              alt="LinkedIn"
              class="sc-jKJlTe cxGEaI"
              src="Icon-linkedin.png"
            />
            <img
              alt="Twitter"
              class="sc-jKJlTe cxGEaI"
              src="Icon-twitter.png"
            />
            <img
              alt="Facebook"
              class="sc-jKJlTe cxGEaI"
              src="Icon-facebook.png"
            />
          </div>
          <div>
            <a
              class="sc-kEYyzF sc-iAyFgw bHoBLw"
              href="https://hackton.co/#team"
            >
              Team
            </a>
            <a
              class="sc-kEYyzF sc-iAyFgw bHoBLw"
              href="#"
            >
              Privacy
            </a>
            <a
              class="sc-kEYyzF sc-iAyFgw bHoBLw"
              href="#"
            >
              Contacts
            </a>
          </div>
          <div>
            <p
              class="sc-jzJRlG sc-cSHVUG iodlNy"
            >
              International Crafters © 2020
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
